src/prefect/flows.py

```diff
@@ -158,6 +159,26 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hook passed is list and contains callables
+        hook_types = [on_completion, on_failure, on_cancellation, on_crashed]
```

Comment:
I hate to be a stickler *but* since we're doing duck typing below it seems a tad confusing to have `_types` in this variable (to me). I might suggest `hook_categories` or even `hook_kinds`🤷‍♀️

Reply:
Yeah sure, changed it to `hook_categories`